collect.py

```diff
@@ -119,25 +150,24 @@ def do_something(driver, elem_attributes=None):
             if not child.is_displayed() or not child.is_enabled():
                 continue
 
-            elem = child
-            break
+            res = check_attributes(driver, elem_attributes, child)  # returns list of elements with attributes = child_attributes
```

Comment:
Pass `children` as a parameter to this function, so you don't have to find the elements again.